### PR TITLE
NextOffet Value is Handled for Retrieve Subscriptions API

### PR DESCRIPTION
**Pull request by cb-gadagoju**
Overview:
Next Offset value is not handled for `retrieveSubscriptions` Method 
in this Pr we handled it and added a new public method  `retrieveSubscriptionsList` since the return handler is different from previous one could not able to mark it as deprecated on the existing method name `retrieveSubscriptions` so added new method name.
Changes made in  files `CBSubscriptionManager` and `Chargebee` and updated example App with the latest method.

### CODE REVIEW

File: Chargebee/Classes/Configuration/Chargebee.swift [hunk 1]
```diff
@@ -39,10 +39,20 @@ public class Chargebee {
...
+    @available(*, deprecated, message: "This will be removed in upcoming versions, Please use this API func retrieveSubscriptionsList(queryParams: [String: String]? = nil, handler: @escaping RetrieveSubscriptionHandler)")
     public func retrieveSubscriptions(queryParams: [String: String]? = nil, handler: @escaping SubscriptionHandler) {
         let logger = CBLogger(name: "Subscription", action: "Fetch Subscription using customerId")
         logger.info()
+        
+        let request = CBAPIRequest(resource: SubscriptionResource(queryParams: queryParams))
+        CBSubscriptionManager().retrieveSubscriptions(network: request, logger: logger, handler: handler)
+    }
 
+    public func retrieveSubscriptionsList(queryParams: [String: String]? = nil, handler: @escaping RetrieveSubscriptionHandler) {
```
Comment (cb-haripriyan) [inline Chargebee/Classes/Configuration/Chargebee.swift:52]: Why can we not use `retrieveSubscriptions`?

File: Chargebee/Classes/Subscription/CBSubscription.swift [hunk 2]
```diff
@@ -85,6 +87,21 @@ class CBSubscriptionManager {
...
     }
 
+    func retrieveSubscriptions<T: CBNetworkRequest>(network: T, logger: CBLogger, handler: @escaping RetrieveSubscriptionHandler) {
+        let (onSuccess, onError) = CBResult.buildResultHandlers(handler, logger)
+        network.load(withCompletion: { status in
+            if let data = status as? CBSubscriptionWrapper {
+                if data.list.isEmpty {
+                    onError(CBError.defaultSytemError(statusCode: 404, message: "Subscription Not found"))
+                }else {
+                    onSuccess(CBSubscriptionWrapper.init(list: data.list, nextOffset: data.nextOffset))
```
[the remark this answers is not in the document]
Comment (cb-gadagoju) [reply Chargebee/Classes/Subscription/CBSubscription.swift:97]: Earlier we have only SubscriptionList  but now offset value also we have to pass ,so i created a struct  CBSubscriptionWrapper and the handler (`(CBResult<CBSubscriptionWrapper>)`)is expecting CBSubscriptionWrapper.

File: Chargebee/Classes/Configuration/Chargebee.swift (hunk 1 above)
Comment (cb-haripriyan) [reply Chargebee/Classes/Configuration/Chargebee.swift:52]: Can you please verify this again?

File: Chargebee/Classes/Subscription/CBSubscription.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (cb-haripriyan) [reply Chargebee/Classes/Subscription/CBSubscription.swift:97]: Please verify again. the type of `data` is already `CBSubscriptionWrapper`

**Review by cb-haripriyan — CHANGES_REQUESTED** [review]

File: Chargebee/Classes/Subscription/CBSubscription.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (cb-gadagoju) [reply Chargebee/Classes/Subscription/CBSubscription.swift]: > Please verify again. the type of `data` is already `CBSubscriptionWrapper`

yes update data is same type!

File: Chargebee/Classes/Configuration/Chargebee.swift (hunk 1 above)
Comment (cb-gadagoju) [reply Chargebee/Classes/Configuration/Chargebee.swift]: due to some problem in Xcode it didn't reflected in my machine but now its working as expected refactored the name and reverted back with existing name  `retrieveSubscriptions` Thanks !

File: Chargebee/Classes/Configuration/Chargebee.swift [hunk 3]
```diff
@@ -39,10 +39,20 @@ public class Chargebee {
...
         logger.info()
+        
+        let request = CBAPIRequest(resource: SubscriptionResource(queryParams: queryParams))
+        CBSubscriptionManager().retrieveSubscriptions(network: request, logger: logger, handler: handler)
+    }
 
+    public func retrieveSubscriptions(queryParams: [String: String]? = nil, handler: @escaping RetrieveSubscriptionHandler) {
+        let logger = CBLogger(name: "Subscription", action: "Fetch Subscription using customerId")
+        logger.info()
+
```
Comment (cb-haripriyan) [inline Chargebee/Classes/Configuration/Chargebee.swift:55]: Can we update the README as well?

**Review by cb-haripriyan — APPROVED** [review]

**Pull request merged**